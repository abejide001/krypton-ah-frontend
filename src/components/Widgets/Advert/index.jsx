import React from 'react';
import { Grid, Image } from 'semantic-ui-react';

export default function index() {
  return (
    <div>
      <Grid.Column className="Advert">
        <Image src="https://res.cloudinary.com/ah-krypton/image/upload/v1548328425/mm17kbfqf7sjrw2w45rq.png" />
      </Grid.Column>
    </div>
  );
}
