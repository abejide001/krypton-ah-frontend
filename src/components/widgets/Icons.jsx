import React from 'react';
import PropTypes from 'prop-types';
import { Menu, Icon, Rating } from 'semantic-ui-react';
import Reactions from './ReactionIcons';

export default function Icons(props) {
  const { id, ratingIcon, numberofcomments, date, name, rating, disabled } = props;
  return (
    <Menu className="widgetContainer">
      <div className="icons">
        {ratingIcon ? (
          <React.Fragment>
            <Reactions date={date} selectedArticleId={id} />
          </React.Fragment>
        ) : (
          <React.Fragment>
            <Reactions date={date} />
            <Icon disabled link size="small" name="bookmark outline" />
            <span>{numberofcomments}</span>
            <div className="rating">
              <Rating disabled={disabled} maxRating={5} defaultRating={rating} size="small" />
              {name ? <span>{name}</span> : ''}
            </div>
          </React.Fragment>
        )}
      </div>
    </Menu>
  );
}
Icons.defaultProps = {
  id: 1,
  name: 'author-name',
  date: 'current-date',
  ratingIcon: 'rating-class',
  numberofcomments: 23,
  rating: 0,
  disabled: false
};
Icons.propTypes = {
  id: PropTypes.number,
  name: PropTypes.string,
  date: PropTypes.string,
  ratingIcon: PropTypes.string,
  numberofcomments: PropTypes.number,
  rating: PropTypes.number,
  disabled: PropTypes.bool
};
