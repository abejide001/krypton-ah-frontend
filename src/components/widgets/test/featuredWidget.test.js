import React from 'react';
import { shallow } from 'enzyme';
import Widget from '../Articlewidgets/ArticlesFeedContainer';
import Advert from '../Advert';
import Icons from '../ReactionIcons';

describe('button test', () => {
  it('should test for the button component', () => {
    const descriptionProps = 'this is a test description';
    const classNameProps = 'className';
    const wrapper = shallow(<Widget description={descriptionProps} className={classNameProps} />);
    expect(wrapper.find('Feed')).toBeDefined();
    expect(wrapper.find('featuredwidget')).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });

  it('should test for the advert component', () => {
    const wrapper = shallow(<Advert />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should test for the Icons component', () => {
    const wrapper = shallow(<Icons />);
    expect(wrapper).toMatchSnapshot();
  });
});
