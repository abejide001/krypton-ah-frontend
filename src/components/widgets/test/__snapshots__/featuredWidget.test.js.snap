// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button test should test for the Icons component 1`] = `
<div>
  <i
    className="dates"
  >
    28th february 2019
  </i>
  <i
    onClick={[Function]}
    role="presentation"
  >
    <Icon
      as="i"
      disabled={true}
      fitted={true}
      link={true}
      name="thumbs up thumbs up outline"
      size="small"
    />
  </i>
  23
  <i
    onClick={[Function]}
    role="presentation"
  >
    <Icon
      as="i"
      disabled={true}
      fitted={true}
      link={true}
      name="thumbs down thumbs down outline"
      size="small"
    />
  </i>
  23
  <i>
    <Icon
      as="i"
      disabled={true}
      fitted={true}
      link={true}
      name="comments outline"
      size="small"
    />
  </i>
  23
</div>
`;

exports[`button test should test for the advert component 1`] = `
<GridColumn
  className="Advert"
>
  <Image
    as="img"
    className="AdvertImg"
    src="https://res.cloudinary.com/ah-krypton/image/upload/v1548328425/mm17kbfqf7sjrw2w45rq.png"
    ui={true}
  />
</GridColumn>
`;

exports[`button test should test for the button component 1`] = `
<Card
  className="widgetIcons"
>
  <CardContent
    className="className"
  >
    <Feed>
      <MainFeaturedArticle
        description="placeholder descritption"
        image="https://res.cloudinary.com/ah-krypton/image/upload/v1550010452/widget.png"
        title="placeholder title"
      />
    </Feed>
  </CardContent>
</Card>
`;
